A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/FMC/A32NX_FMCMainDisplay.js

```diff
@@ -3120,6 +3126,246 @@ class FMCMainDisplay extends BaseAirliners {
         }
     }
 
+     /**
+     * Check if a place is the correct format for a runway
+     * @param {string} place
+     */
+    isRunwayFormat(place) {
+        return place.match(/^([A-Z]{4})([0-9]{2}[RCL]?)$/) !== null;
+    }
+
+    /**
+     * Parse a runway string and return the location of the threshold
+     * Returns undefined if invalid format or not in database
+     * @param {string} place
+     * @param {function} callback Param is either undefined or LatLongAlt
+     */
+    parseRunway(place, callback) {
+        const rwy = place.match(/^([A-Z]{4})([0-9]{2}[RCL]?)$/);
+        if (rwy !== null) {
+            this.dataManager.GetAirportByIdent(rwy[1]).then((airport) => {
+                if (airport) {
+                    for (let i = 0; i < airport.infos.oneWayRunways.length; i++) {
+                        if (Avionics.Utils.formatRunway(airport.infos.oneWayRunways[i].designation) === rwy[2]) {
+                            const runway = airport.infos.oneWayRunways[i];
+                            // this should be to the treshold but we don't have that, so we just use half the length
+                            const adjustedCoordinates = Avionics.Utils.bearingDistanceToCoordinates((runway.direction + 180) % 360, runway.length / 2 * 0.000539957, runway.latitude, runway.longitude); // TODO proper m to nm conversion
```

Comment:
Do we have unit conversions somewhere? I haven't found anything but I think it would be useful to have a little util class somewhere. Maybe I can add it if it doesn't exist.